tests/integ/test_reduce_describe_query.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import List
+
+import pytest
+
+from snowflake.snowpark._internal.analyzer.expression import Attribute
+from snowflake.snowpark.functions import col
+from snowflake.snowpark.session import (
+    _PYTHON_SNOWPARK_REDUCE_DESCRIBE_QUERY_ENABLED,
+    Session,
+)
+from tests.integ.utils.sql_counter import SqlCounter
+from tests.utils import IS_IN_STORED_PROC
+
+pytestmark = [
+    pytest.mark.skipif(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="Reducing describe queries is not supported in Local Testing",
+    ),
+]
+
+
+def setup(session):
+    is_reduce_describe_query_enabled = session.reduce_describe_query_enabled
+    session.reduce_describe_query_enabled = True
+    yield
+    session.reduce_describe_query_enabled = is_reduce_describe_query_enabled
+
+
+# TODO SNOW-1728988: enable test cases with select after caching attributes on SelectStatement
+# Create from SQL
+create_from_sql_funcs = [
+    lambda session: session.sql("SELECT 1 AS a, 2 AS b"),
+    # lambda session: session.sql("SELECT 1 AS a, 2 AS b").select("b"),
```

Comment:
remove commented code?

Reply:
I think @sfc-gh-jdu want to enable this when the inferring also works on selectStatement, but @sfc-gh-jdu i thought this works when sql simplifier is enabled, maybe we can only disable those case on sql simplifier is enabled to be more clear

Reply:
no worries, I think I can remove them for now, and add them back in next PR (very soon)